### PR TITLE
Remove expressible as syntax token

### PR DESCRIPTION
**Pull request by kimdv**
Depends on #339

### CODE REVIEW

**Review by ahoppen — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by kimdv** [comment]
@swift-ci Please test

File: Sources/SwiftSyntaxBuilder/Buildables.swift.gyb
Comment (kimdv) [inline Sources/SwiftSyntaxBuilder/Buildables.swift.gyb:278]: @ahoppen added this section after your approval

File: Sources/SwiftSyntaxBuilder/Buildables.swift.gyb
Comment (kimdv) [inline Sources/SwiftSyntaxBuilder/Buildables.swift.gyb:277]: @ahoppen added this section after your approval. 

I also discovered, in another PR, that if we don't add `[:]` ad the end, we are copy by reference. 
I'm not sure why it workers the other places. 

Correct me if I'm wrong

Comment (ahoppen) [reply Sources/SwiftSyntaxBuilder/Buildables.swift.gyb:277]: Oh, that is surprising. Very good catch.

To make sure we don’t hit this again, I think it would be good if we wrapped `SYNTAX_COLLECTION_EXPRESSIBLE_AS_CONFORMANCES.get` in a function. 
E.g.
```python
def get_expressible_as_conformances(conforming_type):
  expressible_as_protocols = (SYNTAX_COLLECTION_EXPRESSIBLE_AS_CONFORMANCES.get(conforming_type) or [])[:]
  expressible_as_protocols += (SYNTAX_BUILDABLE_EXPRESSIBLE_AS_CONFORMANCES.get(conforming_type) or [])[:]
  return map(lambda x : 'ExpressibleAs' + x, expressible_as_protocols)
```

and also use that in the other places.

Comment (kimdv) [reply Sources/SwiftSyntaxBuilder/Buildables.swift.gyb]: Something like this?

**Comment by kimdv** [comment]
@swift-ci Please test

File: Sources/SwiftSyntaxBuilder/Buildables.swift.gyb
Comment (ahoppen) [inline Sources/SwiftSyntaxBuilder/Buildables.swift.gyb:13]: Could you add a comment why `[:]` is necessary here? Something like
```
# Use [:] to make sure we copy the conformances so that we don't modify `SYNTAX_COLLECTION_EXPRESSIBLE_AS_CONFORMANCES` when adding elements to `expressible_as_protocols` below.
```

**Review by ahoppen — APPROVED** [review]
One request for a comment, otherwise looking good.

**Comment by ahoppen** [comment]
🎉 CI testing is honoring your requests, @kimdv 🎉

**Comment by kimdv** [comment]
@swift-ci Please test

**Comment by kimdv** [comment]
@ahoppen I have added the comment and triggered the CI. 
I assume that I can merge, now that you have approved it? 

If not, sorry. Please tell then!

**Comment by ahoppen** [comment]
Yes, if CI passes and you’ve got approval, you may merge.

**Pull request merged**
